core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_utils/RangeMapUtils.java

```diff
@@ -0,0 +1,70 @@
+package fr.sncf.osrd.envelope_utils;
+
+import com.google.common.collect.Range;
+import com.google.common.collect.RangeMap;
+import com.google.common.collect.TreeRangeMap;
+import java.util.function.BiFunction;
+
+public class RangeMapUtils {
+    /**
+     * Returns a range map where the adjacent ranges of same values have been merged
+     */
+    public static <T> RangeMap<Double, T> mergeRanges(RangeMap<Double, T> map) {
+        TreeRangeMap<Double, T> result = TreeRangeMap.create();
+        var entryIterator = map.asMapOfRanges().entrySet().iterator();
+        if (!entryIterator.hasNext())
+            return result;
+        var currentEntry = entryIterator.next();
+        var currentRange = currentEntry.getKey();
+        var currentValue = currentEntry.getValue();
+        while (entryIterator.hasNext()) {
+            var nextEntry = entryIterator.next();
+            var nextRange = nextEntry.getKey();
+            var nextValue = nextEntry.getValue();
+            if (currentValue.equals(nextValue) && currentRange.isConnected(nextRange)) {
+                currentRange = Range.closedOpen(currentRange.lowerEndpoint(), nextRange.upperEndpoint());
+            } else {
+                result.put(currentRange, currentValue);
+                currentRange = nextRange;
+                currentValue = nextValue;
+            }
+        }
+        result.put(currentRange, currentValue);
```

Comment:
Can't do that, you won't merge the 2 last ranges even if they have the same value

Reply:
I really think the algorithm works, I'll write a test to enforce it

Reply:
My bad, I think the algo is working ! A test would be super nice :)